api/deploy/ocp/scc.go

```
@@ -0,0 +1,103 @@
+// artifacts with defails values.
+var SecurityContextConstraintsDefaults = SecurityContextConstraintsValues{
+	Namespace: "ceph-csi",
+	Deployer:  "",
+}
+
+// NewSecurityContextConstraints creates a new SecurityContextConstraints
+// object by replacing variables in the template by the values set in the
+// SecurityContextConstraintsValues.
+//
+// The deployer parameter (when not an empty string) is used as a prefix for
+// the name of the SCC and the linked ServiceAccounts.
+func NewSecurityContextConstraints(values SecurityContextConstraintsValues) *secv1.SecurityContextConstraints {
+	data := NewSecurityContextConstraintsYAML(values)
+
+	scc := &secv1.SecurityContextConstraints{}
+	err := yaml.Unmarshal([]byte(data), scc)
+	if err != nil {
+		panic(err)
+	}
+
+	return scc
+}
+
+// internalSecurityContextConstraintsValues extends
+// SecurityContextConstraintsValues with some private attributes that may get
+// set based on other values.
+type internalSecurityContextConstraintsValues struct {
+	SecurityContextConstraintsValues
+
+	// Prefix is based on SecurityContextConstraintsValues.Deployer.
+	Prefix string
+}
+
+// NewSecurityContextConstraintsYAML returns a YAML string where the variables
+// in the template have been replaced by the values set in the
+// SecurityContextConstraintsValues.
+func NewSecurityContextConstraintsYAML(values SecurityContextConstraintsValues) string {
+	var buf bytes.Buffer
+
+	// internalValues is a copy of values, but will get extended with
+	// API-internal values
+	internalValues := internalSecurityContextConstraintsValues{
+		SecurityContextConstraintsValues: values,
+	}
+
+	if internalValues.Deployer != "" {
+		internalValues.Prefix = internalValues.Deployer + "-"
+	}
+
+	tmpl, err := template.New("SCC").Parse(securityContextConstraints)
+	if err != nil {
+		panic(err)
+	}
+	err = tmpl.Execute(&buf, internalValues)
+	if err != nil {
+		panic(err)
+	}
```

should we return the error let the caller handle the error message?

---

Not sure how this can fail?

---

Yeah even not sure, but let's not panic and return the error message let the caller handle it?